samples/cellular/battery/src/main.c

```diff
@@ -89,7 +89,7 @@ static void low_level_state_update(struct k_work *item)
 	k_sem_take(&state_sem, K_FOREVER);
 
 	if (state == NORMAL_HIGH_BAT) {
-		err = lte_lc_deinit();
+		err = lte_lc_power_off();
 		if (err) {
 			printk("Modem deinitialization failed, err: %d\n", err);
```

Comment:
If we want to rephrase that, I would change it to something that indicates that the CFUN operation has failed, e.g.
`Failed to switch to functional mode 0` or `Failed to CFUN=0`
The modem is happily "on", as in powered up and running.

Reply:
Maybe `Modem power off mode failed`, to align with the other prints in this sample e.g.
`printk("Modem RX only mode failed, err: %d\n", err);`
`printk("Modem normal mode failed, err: %d\n", err);`

Reply:
I'm not following. If you call a function which is named `power_off()` and it returns an error, it should be correct to say that "powering off failed", I think. 😄

Reply:
Well, I see your point.. but the "power off" terminology while being "AT correct" doesn't fit too well when the modem is not just an AT box. Saying that the modem is "powered off" just doesn't sound very correct: the modem SoC is powered on, you can send AT commands, provision certificates, even do a firmware update.. it is emitting traces, the application can communicate with it.

I would like to make it clear that the modem is not "off" in that sense, but in the "AT" sense, that is, it is in minimum functional mode.

Reply:
I think we should aim to keep the terminology aligned between nRF9x Product Specification, the AT command documentation and the SDK. The AT command documentation does talk about both "power off" and "minimal functionality", so there would be room for improvement. I had a look at the nRF9160 Product Specification and it uses the term "modem off", for example here: [Sleep](https://infocenter.nordicsemi.com/topic/ps_nrf9160/_tmp/alta.nRF9160/autodita/CURRENT/parameters.id_current_sleep.html)

I'm not too familiar with the power management in the SiP. Obviously modem SoC is not totally powered off with `AT+CFUN=0`, because it does respond to AT commands. But, when the MCU is not sending AT commands to the modem, there are no traces in this state either.

There are registers `LTEMODEM.STARTN` and `LTEMODEM.FORCEOFF`, but there's also the following note: _Starting and stopping LTE modem must only be done through the LTE modem API to guarantee correct sequence in FW and HW and to avoid possible malfunctions._ What does the "LTE modem API" refer to? The AT command interface or libmodem, I'm not sure.

But, if `AT+CFUN=0` puts the modem SoC into the state with the lowest possible current consumption (without powering off the whole system), I think it's warranted to talk about modem power off.

Reply:
I have made the change as you suggested, so we can go ahead with this.

But I remain of the opinion that the term "modem" is overloaded: the "modem SoC" is one thing, the "LTE modem" is another. While in the AT specification "modem" refers to the "LTE modem", in NCS we also use the term "modem" to refer to the "modem SoC" and the two things are distinct.

It should be clear which is which when we talk about "modem", we cannot talk about the Modem as the LTE Modem in all contexts. When it comes to things like modem firmware updates, they are much harder to document if you can't talk about the modem SoC, i.e. "to apply a firmware update you need to power off the modem" -> `CFUN=0` that is not what we mean right? We mean power off the SoC.

Reply:
I agree the terminology is confusing and even wrong but having "power off" in this log is aligned to what is done elsewhere in the documentation and code. The changes should be done throughout if we wanted to change it.

Reply:
Yes, it has become evident that there's room for improvement. I have discussed this with the modem team and at least the AT command documentation needs an update. And when the `+CFUN=0` documentation is updated, we need to consider about updating the LTE link control API as well.